backtesting/backtesting.py

```diff
@@ -115,13 +117,21 @@ def I(self,  # noqa: E743
             def init():
                 self.sma = self.I(ta.SMA, self.data.Close, self.n_sma)
         """
+        def _format_name(name: str) -> str:
+            return name.format(*map(_as_str, args),
+                               **dict(zip(kwargs.keys(), map(_as_str, kwargs.values()))))
+
         if name is None:
             params = ','.join(filter(None, map(_as_str, chain(args, kwargs.values()))))
             func_name = _as_str(func)
             name = (f'{func_name}({params})' if params else f'{func_name}')
+        elif isinstance(name, str):
+            name = _format_name(name)
+        elif try_(lambda: all(isinstance(item, str) for item in name), False):
+            name = [_format_name(item) for item in name]
         else:
-            name = name.format(*map(_as_str, args),
-                               **dict(zip(kwargs.keys(), map(_as_str, kwargs.values()))))
+            raise TypeError(f'Unexpected `name` type {type(name)}, `str` or `Iterable[str]` '
+                            'was expected.')
```

Comment:
```suggestion
            raise TypeError(f'Unexpected `name=` type {type(name)}; expected `str` or `Sequence[str]`')
```
`Iterable` such as a generator would get expent by our `isinstance()` checks.